### PR TITLE
Version store documentation

### PR DESCRIPTION
**Pull request by bmoscon**
Create initial first pass of version store documentation. I've tried to cover the most important use cases for the main methods of version store, focusing on the basic usecase (dataframes). Is surely not complete, but I think its much better than what we currently have (nothing) :)

I will squash the commits into a single commit

### CODE REVIEW

**Comment by bmoscon** [comment]
@dimosped @jamesblackburn @TomTaylorLondon please have a look. Feel free to add anyone else to the review that may want to look it over

**Comment by bmoscon** [comment]
its probably easiest to read it over here: https://github.com/manahl/arctic/blob/version-store-docs/docs/versionstore.md

File: docs/versionstore.md [hunk 1]
```diff
@@ -1,3 +1,316 @@
...
\ No newline at end of file
+
+(note: current doc is based on arctic-1.68.0)
+
+VersionStore serializes and stores Pandas objects, numpy arrays as well as other python types in MongoDB. Objects are `versioned` and new versions are created when a `symbol` is modified. 
+
+
+# Reading and Writing Data with VersionStore
+
+```
```
Comment (dimosped) [inline docs/versionstore.md]: not sure if specifying the language makes any difference, checking:

python defined:
```python 
import lz4

def print_lz4_version():
    print(lz4.__version__)
```

vs 

default lang:
```
import lz4

def print_lz4_version():
    print(lz4.__version__)
```

Comment (dimosped) [reply docs/versionstore.md]: basically its colors for syntax highlighting

**Comment by dimosped** [comment]
great @bmoscon , the vstore documentation looks quite complete.
If we get readthedocs integration, we can add links to API documentation.

File: docs/versionstore.md (hunk 1 above)
Comment (bmoscon) [reply docs/versionstore.md]: the dataframe output and the >>> tend to not render well when you use the `python` formatting

**Pull request merged**